src/util/taproot.rs

```diff
@@ -82,12 +93,804 @@ sha256t_hash_newtype!(TapSighashHash, TapSighashTag, MIDSTATE_TAPSIGHASH, 64,
     doc="Taproot-tagged hash for the taproot signature hash", true
 );
 
+impl TapTweakHash {
+
+    /// Create a new BIP341 [`TapTweakHash`] from key and tweak
+    /// Produces H_taptweak(P||R) where P is internal key and R is the merkle root
+    pub fn from_key_and_tweak(
+        internal_key: schnorr::PublicKey,
+        merkle_root: Option<TapBranchHash>,
+    ) -> TapTweakHash {
+        let mut eng = TapTweakHash::engine();
+        // always hash the key
+        eng.input(&internal_key.serialize());
+        if let Some(h) = merkle_root {
+            eng.input(&h);
+        } else {
+            // nothing to hash
+        }
+        TapTweakHash::from_engine(eng)
+    }
+}
+
+impl TapLeafHash {
+    /// function to compute leaf hash from components
+    pub fn from_script(script: &Script, ver: LeafVersion) -> TapLeafHash {
+        let mut eng = TapLeafHash::engine();
+        ver.as_u8()
+            .consensus_encode(&mut eng)
+            .expect("engines don't err");
```

Comment:
nit:
```suggestion
            .expect("TapLeafHash engine broken on tapleaf version serialization");
```

Reply:
Honestly I'd rather this be an `.unwrap()` with a comment. All these unique strings will eventually add up to nontrivial library bloat, for code which will literally never be exercised.